src/components/Header.tsx

```diff
@@ -1,32 +1,44 @@
 import logo from '../image/logo.png';
 
+import { useAuth } from './hooks/useProvideAuth';
+
 interface IHeader {
   onSignInBtnClick: () => void;
   onSignUpBtnClick: () => void;
 }
 
 export function Header({ onSignInBtnClick, onSignUpBtnClick }: IHeader) {
+  const auth = useAuth();
   return (
     <header className="h-28 w-4/5 m-auto">
       <div className="h-full flex items-center">
         <div className="w-full flex justify-between items-center">
           <a href="/">
             <img width={80} src={logo} alt="main-logo" />
           </a>
-          <div className="w-36 flex justify-between">
+          {auth.userData?.user ? (
             <button
               className="text-green-300 hover:text-green-700 cursor-pointer"
               onClick={onSignInBtnClick}
             >
-              Sign In
-            </button>
-            <button
-              className="text-green-300 hover:text-green-700 cursor-pointer"
-              onClick={onSignUpBtnClick}
-            >
-              Sign Up
+              {auth.userData.user.email}
             </button>
```

Comment:
Here we can make a UI like on airbnb or medium: an avatar and a menu which opens when you click on the avatar. And a signOut button inside the menu. https://tppr.me/voAj5